Review comments and fixes

ritwikbasu opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR -->

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

ritwikbasu commented:
This is the same PR as https://github.com/Azure/azure-powershell/pull/7106, but created against Ignite release branch.

praries880 commented on src/ResourceManager/Compute/Commands.Compute.Test/Commands.Compute.Test.csproj:
```diff
@@ -46,8 +46,8 @@
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Network, Version=19.0.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.19.3.0-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
+    <Reference Include="Microsoft.Azure.Management.Network, Version=20.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
tqalk to @MikhailTryakhov  about the version, I think the latest version he is releasing is 20.0.0. All the PRs going into the September branch need to use that version

ritwikbasu replied:
@praries880 - Yes that is correct, I will update this PR when the version is available.

praries880 replied:
.NET SDK is merged and (with a bit of magic) finally published https://www.nuget.org/packages/Microsoft.Azure.Management.Network/19.4.0-preview 

So now all contributors must change SDK version from local version (localFeed folder) to 19.4.0 and remove nuget packages from localFeed folder.

maddieclayton commented on src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/CortexTests.ps1:
```diff
@@ -0,0 +1,127 @@
+	finally
+	{
+		$deleteVpnGateway = Remove-AzureRmVpnGateway -ResourceGroupName $rgName -Name $vpnGatewayName -Force -Passthru
+		Assert-AreEqual true $deleteVpnGateway
```
Also add a test for Remove-AzureRmVpnConnection

ritwikbasu replied:
Added

maddieclayton commented on src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/CortexTests.ps1:
```diff
@@ -0,0 +1,127 @@
+		$createdVirtualWan = Update-AzureRmVirtualWan -ResourceGroupName $rgName -Name $virtualWanName -AllowVnetToVnetTraffic $false -AllowBranchToBranchTraffic $false
+		$virtualWan = Get-AzureRmVirtualWan -ResourceGroupName $rgName -Name $virtualWanName
+		Assert-AreEqual $rgName $virtualWan.ResourceGroupName
+		Assert-AreEqual $virtualWanName $virtualWan.Name
```
None of the asserts in the test are testing the values of the properties that you are explicitly setting, so it is hard to tell whether the Update cmdlet is working.  Please add more asserts for all the cmdlets.

ritwikbasu replied:
Fixed

maddieclayton commented on src/ResourceManager/Network/Commands.Network/Az.Network.psd1:
```diff
@@ -337,7 +337,35 @@ CmdletsToExport = 'Add-AzApplicationGatewayAuthenticationCertificate',
+    'Remove-AzVpnConnection',
+    'New-AzVirtualHubVnetConnection',
+    'Get-AzVirtualHubVnetConnection',
+    'Update-AzVirtualHubVnetConnection',
```
think you removed this.

ritwikbasu replied:
Fixed.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/AzureRM.Network.psd1:
```diff
@@ -338,7 +338,34 @@ CmdletsToExport = 'Add-AzureRmApplicationGatewayAuthenticationCertificate',
+               'Remove-AzureRmVpnConnection',
+               'New-AzureRmVirtualHubVnetConnection',
+               'Get-AzureRmVirtualHubVnetConnection',
+               'Update-AzureRmVirtualHubVnetConnection',
```
Same - think you deleted this cmdlet.

ritwikbasu replied:
Fixed

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/AzureRM.Network.md:
```diff
@@ -63,7 +63,7 @@ Adds a backend address pool configuration to a load balancer.
 Adds a front-end IP configuration to a load balancer.
 
 ### [Add-AzureRmLoadBalancerInboundNatPoolConfig](Add-AzureRmLoadBalancerInboundNatPoolConfig.md)
-Adds an inbound NAT pool configuration to a load balancer.
```
ping - please add descriptions back that are missing.

ritwikbasu replied:
Auto generation removes these references every time I auto generate help. Seems some issue with the source repo. These are not my changes.

ritwikbasu replied:
Auto generation help is removing these entries and is adding entries that do not exist in the original repo and are not part of my PR. There is some issue with the help files in the original repo. Fixing these is beyond the scope of this PR.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/AzureRM.Network.md:
```diff
@@ -86,6 +86,9 @@ Adds a route to a route table.
 Adds a route filter rule to a route filter.
 
+### [Add-AzureRmServiceEndpointPolicyDefinition](Add-AzureRmServiceEndpointPolicyDefinition.md)
+{{Fill in the Synopsis}}
```
Please fill in any missing synopsises in this file

ritwikbasu replied:
These are not my changes and beyond my purview. It seems someone added these files without checking in the changes to azurerm.network md.

ritwikbasu replied:
This is auto generated and beyond the scope of this PR - I am not sure what this command is.

ritwikbasu replied:
Looks like in the original repo the help files/cmdlets are there without network.md update or the other way round. This is causing this error - made this file have only changes related to this PR.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/Get-AzureRmVirtualHub.md:
````diff
@@ -0,0 +1,133 @@
+
+### ListBySubscriptionId
+```
+Get-AzureRmVirtualHub [-ResourceGroupName <String>] [-Name <String>] [-DefaultProfile <IAzureContextContainer>]
````
This should have no parameters, rather than both resource group name and name optional.  Apply to all gets.

ritwikbasu replied:
This is auto generated. I used two parameter sets one for sub id and one for RG name. I am not sure how to have no parameters.

maddieclayton replied:
Answered this question in the cmdlets themselves.  Please be sure to regenerate all files to get this change in.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/Get-AzureRmVirtualWanVpnConfiguration.md:
```diff
@@ -0,0 +1,255 @@
+
+## SYNTAX
+
+### ByVirtualWanObjectByVpnSiteObject (Default)
```
Default should always be the interactive parameter set (using names).  Apply everywhere.

ritwikbasu replied:
Fixed

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/Get-AzureRmVirtualWanVpnConfiguration.md:
````diff
@@ -0,0 +1,255 @@
+PS C:\> $vpnSitesForConfig = New-Object Microsoft.Azure.Commands.Network.Models.PSVpnSite[] 1
+PS C:\> $vpnSitesForConfig[0] = $vpnSite
+PS C:\> Get-AzureRmVirtualWanVpnConfiguration -VirtualWan $virtualWan -StorageSasUrl "SignedSasUrl" -VpnSite $vpnSitesForConfig
+```
````
output?

ritwikbasu replied:
There is no output for this - download will write the file to the blob provided as part of the commandlet.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/Get-AzureRmVirtualWan.md:
```diff
@@ -0,0 +1,127 @@
+
+The above snippet produces the following output:
+
+Name                       : testRG
```
Please put all of the output inside the ``` block so it is formatted nicely. Apply everywhere

ritwikbasu replied:
Fixed.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/New-AzureRmVirtualHub.md:
````diff
@@ -0,0 +1,330 @@
+```
+New-AzureRmVirtualHub -ResourceGroupName <String> -Name <String> -VirtualWan <PSVirtualWan>
+ -AddressPrefix <String> -Location <String>
+ [-HubVnetConnection <System.Collections.Generic.List`1[Microsoft.Azure.Commands.Network.Models.PSHubVirtualNetworkConnection]>]
````
-HubNetConnection should be an array.

ritwikbasu replied:
Already fixed in code. Regenerated help.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/New-AzureRmVirtualHubRoute.md:
````diff
@@ -0,0 +1,95 @@
+## SYNTAX
+
+```
+New-AzureRmVirtualHubRoute -AddressPrefix <System.Collections.Generic.List`1[System.String]>
````
array, not list

ritwikbasu replied:
Regenerated help

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/New-AzureRmVirtualHubRouteTable.md:
````diff
@@ -0,0 +1,86 @@
+
+```
+New-AzureRmVirtualHubRouteTable
+ -Route <System.Collections.Generic.List`1[Microsoft.Azure.Commands.Network.Models.PSVirtualHubRoute]>
````
array, not list

ritwikbasu replied:
Already fixed in code - regenerated help.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/New-AzureRmVirtualHubRouteTable.md:
```diff
@@ -0,0 +1,86 @@
+PS C:\> $routeTable = New-AzureRmVirtualHubRouteTable -Route @($route1, $route2)
+PS C:\> New-AzureRmResourceGroup -Location "West US" -Name "testRG"
+PS C:\> $virtualWan = New-AzureRmVirtualWan -ResourceGroupName "testRG" -Name "myVirtualWAN" -Location "West US"
+PS C:\> New-AzureRmVirtualHub -VirtualWanId $virtualWan.Id -ResourceGroupName "testRG" -Name "westushub" -AddressPrefix "10.0.1.0/24" -RouteTable $routeTable
```
output?

ritwikbasu replied:
Added

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/New-AzureRmVirtualWan.md:
````diff
@@ -0,0 +1,203 @@
+
+```
+New-AzureRmVirtualWan -Name <String> -ResourceGroupName <String> -Location <String>
+ [-AllowVnetToVnetTraffic <Boolean>] [-AllowBranchToBranchTraffic <Boolean>] [-Tag <Hashtable>] [-AsJob]
````
All "New" cmdlet should have SwitchParameters instead of booleans.  Apply everywhere.

ritwikbasu replied:
Regenerated help - was already fixed in code.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/New-AzureRmVirtualWan.md:
````diff
@@ -0,0 +1,203 @@
+
+```powershell
+PS C:\> New-AzureRmResourceGroup -Location "West US" -Name "testRG" 
+PS C:\> New-AzureRmVirtualWan -ResourceGroupName "testRG" -Name "myVirtualWAN" -Location "West US" -AllowBranchToBranchTraffic $true
````
output? apply everywhere.

ritwikbasu replied:
Added

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/New-AzureRmVpnGateway.md:
````diff
@@ -0,0 +1,279 @@
+
+### ByVirtualHubName (Default)
+```
+New-AzureRmVpnGateway -Name <String> -ResourceGroupName <String> -VpnGatewayScaleUnit <UInt32>
````
Looks like the order on this is wrong.  Please always have "ResourceGroupName" then "<parent>Name" then "Name", then any other parameters.  This depends on the order in your cs file so please update that and then regenerate help.  Apply this for everything - identity parameters should always be first, ordered by top level to bottom level, then all other parameter follow.

ritwikbasu replied:
Regenerated help. Was fixed in code.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/New-AzureRmVpnGateway.md:
````diff
@@ -0,0 +1,279 @@
+```
+New-AzureRmVpnGateway -Name <String> -ResourceGroupName <String> -VpnGatewayScaleUnit <UInt32>
+ -VirtualHubName <String>
+ [-VpnConnection <System.Collections.Generic.List`1[Microsoft.Azure.Commands.Network.Models.PSVpnConnection]>]
````
array not list.  apply to all files.

ritwikbasu replied:
Already fixed in code - regenerated help.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/Remove-AzureRmVirtualWan.md:
````diff
@@ -0,0 +1,229 @@
+```
+
+### -PassThru
+{{Fill PassThru Description}}
````
make sure all these are filled out.

ritwikbasu replied:
Fixed.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/Update-AzureRmVpnGateway.md:
````diff
@@ -0,0 +1,258 @@
+### ByVpnGatewayName (Default)
+```
+Update-AzureRmVpnGateway -Name <String> -ResourceGroupName <String>
+ [-VpnConnection <System.Collections.Generic.List`1[Microsoft.Azure.Commands.Network.Models.PSVpnConnection]>]
````
array not list

ritwikbasu replied:
Was already fixed in code. Regenerated help.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/Update-AzureRmVpnSite.md:
````diff
@@ -0,0 +1,419 @@
+### ByVpnSiteName (Default)
+```
+Update-AzureRmVpnSite -ResourceGroupName <String> -Name <String> [-VirtualWanResourceGroupName <String>]
+ [-VirtualWanName <String>] [-VirtualWan <PSVirtualWan>] [-VirtualWanId <String>] [-IpAddress <String>]
````
If you want to support multiple ways to obtain VirtualWan, you will need to add a set of parameter sets for each.

ritwikbasu replied:
Added parameter sets and updated help module.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/CortexTests.cs:
```diff
@@ -0,0 +1,40 @@
+            XunitTracingInterceptor.AddToContext(new XunitTracingInterceptor(output));
+        }
+
+        [Fact(Skip = "Needs ignite changes to be rolled out to all NRPs.")]
```
Unskipped, added session record, and added note about recording using only Jenkins sub.

ritwikbasu commented:
@maddieclayton , @praries880 - Addressed all review comments, added tests and session records and updated to latest SDK version.

Please take a look.

maddieclayton commented:
@ritwikbasu Will take a look in 30 minutes when the CI finishes (and hopefully passes)! Will include how to fix any build failures as part of the review.

praries880 commented:
@ritwikbasu  credscan issues in the test json : 
C:\workspace\powershell\src\ResourceManager\Network\Commands.Network.Test\SessionRecords\Commands.Network.Test.ScenarioTests.CortexTests\TestCortexCRUD.json

Are the credentials flagged for transient resources that are deleted once the tests are done?

You will need to add suppression for them.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnSite/NewAzureRmVpnSiteCommand.cs:
```diff
@@ -0,0 +1,214 @@
+                vpnSiteToCreate.AddressSpace.AddressPrefixes.AddRange(this.AddressSpace);
+            }
+
+            WriteWarning(string.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
no warnings

ritwikbasu replied:
Changed to WriteVerbose.

maddieclayton replied:
ping

maddieclayton commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnSite/NewAzureRmVpnSiteCommand.cs:
```diff
@@ -0,0 +1,214 @@
+                this.Name,
+                () =>
+                {
+                    WriteWarning(String.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
no warnings

ritwikbasu replied:
Changed to WriteVerbose.

maddieclayton replied:
ping

maddieclayton commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnSite/RemoveAzureRmVpnSiteCommand.cs:
```diff
@@ -0,0 +1,113 @@
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
```
message

ritwikbasu replied:
Fixed

maddieclayton commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnSite/UpdateAzureRmVpnSiteCommand.cs:
```diff
@@ -0,0 +1,411 @@
+            HelpMessage = "The VirtualWan this VpnSite needs to be connected to.")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = CortexParameterSetNames.ByVpnSiteResourceId + "NoVirtualWanUpdate",
```
Confused about this.  Why do you have a separate parameter set for NoVirtualWanUpdate when it is already optional in all other parameter sets? And why are all the VirtualWan related parameters included in this parameter set?

ritwikbasu replied:
So users can select not to update the VirtualWan at all when updating the VpnSite. Not having that was causing the cmdlet to fail with ambiguous parameter set.

maddieclayton replied:
The right way to do this is remove "NoVirtualWanUpdate" from all the parameters related to VirtualWan - currently parameter sets don't make sense.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnSite/UpdateAzureRmVpnSiteCommand.cs:
```diff
@@ -0,0 +1,411 @@
+                    this.Name,
+                    () =>
+                    {
+                        WriteWarning(String.Format(Properties.Resources.UpdatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
no warnings

ritwikbasu replied:
Changed to WriteVerbose

maddieclayton replied:
ping

maddieclayton commented on src/ResourceManager/Network/Commands.Network/help/Get-AzureRmVirtualHub.md:
```diff
@@ -0,0 +1,130 @@
+PS C:\> New-AzureRmVirtualHub -VirtualWan $virtualWan -ResourceGroupName "testRG" -Name "westushub" -AddressPrefix "10.0.1.0/24"
+PS C:\> Get-AzureRmVirtualHub -ResourceGroupName "testRG" -Name "westushub"
+
+Output:
```
don't need this "Output:" line.  Please remove everywhere.

ritwikbasu replied:
Fixed.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/packages.config:
```diff
@@ -2,5 +2,5 @@
   <package id="AutoMapper" version="6.0.2" targetFramework="net452" />
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Network" version="19.3.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.Network" version="19.4.0-preview" targetFramework="net452" />
```
Do you still need this?

ritwikbasu replied:
Yes - these changes are present in the official 19.4 SDK.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/NewAzureRmHubVirtualNetworkConnectionCommand.cs:
```diff
@@ -0,0 +1,193 @@
+            Mandatory = true,
+            HelpMessage = "The resource name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
```
move name below parentresourceid

ritwikbasu replied:
Fixed.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/NewAzureRmHubVirtualNetworkConnectionCommand.cs:
```diff
@@ -0,0 +1,193 @@
+        [Alias("VirtualHubId", "ParentVirtualHubId")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
```
Fixed.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/RemoveAzureRmHubVirtualNetworkConnectionCommand.cs:
```diff
@@ -0,0 +1,140 @@
+            HelpMessage = "Do not ask for confirmation if you want to overwrite a resource")]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false)]
```
Fixed

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/NewAzureRmHubVirtualNetworkConnectionCommand.cs:
```diff
@@ -0,0 +1,193 @@
+                this.Name,
+                () =>
+                {
+                    WriteWarning(String.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
Changed to WriteVerbose

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualHub/RemoveAzureRmVirtualHubCommand.cs:
```diff
@@ -0,0 +1,120 @@
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
```
Fixed.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualHub/NewAzureRmVirtualHubCommand.cs:
```diff
@@ -0,0 +1,164 @@
+                Name,
+                () =>
+                {
+                    WriteWarning(String.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
Replaced with WriteVerbose.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualHub/RemoveAzureRmVirtualHubCommand.cs:
```diff
@@ -0,0 +1,120 @@
+            }
+
+            base.Execute();
+            WriteWarning(Properties.Resources.RemovingVirtualHubWarning);
```
Changed to WriteVerbose

maddieclayton replied:
Ping, this wasn't changed.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualHub/UpdateAzureRmVirtualHubCommand.cs:
```diff
@@ -0,0 +1,161 @@
+                    this.Name,
+                    () =>
+                    {
+                        WriteWarning(String.Format(Properties.Resources.UpdatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
Changed to WriteVerbose

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/GetAzureRmVirtualWanCommand.cs:
```diff
@@ -0,0 +1,63 @@
+        OutputType(typeof(PSVirtualWan))]
+    public class GetAzureRmVirtualWanCommand : VirtualWanBaseCmdlet
+    {
+        [Parameter(
```
Fixed

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/GetAzureRmVirtualWanVpnSitesConfigurationCommand.cs:
```diff
@@ -0,0 +1,166 @@
+                throw new PSArgumentException(Properties.Resources.VirtualWanRequiredForVpnSiteConfiguration);
+            }
+
+            PSVirtualWan virtualWan = this.GetVirtualWan(this.ResourceGroupName, this.Name);
```
No - this is not needed. We would fail the call anyway if it does not.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/NewAzureRmVirtualWanCommand.cs:
```diff
@@ -0,0 +1,116 @@
+        [Parameter(
+           Mandatory = false,
+           HelpMessage = "Allow vnet to vnet traffic for VirtualWan.")]
+        [PSDefaultValue(Help = "$false", Value = false)]
```
Fixed.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/NewAzureRmVirtualWanCommand.cs:
```diff
@@ -0,0 +1,116 @@
+                this.Name,
+                () =>
+                {
+                    WriteWarning(String.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
Changed to WriteVerbose

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/RemoveAzureRmVirtualWanCommand.cs:
```diff
@@ -0,0 +1,111 @@
+           HelpMessage = "Do not ask for confirmation.")]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false)]
```
Fixed.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/SetAzureRmVirtualHubCommand.cs:
```diff
@@ -0,0 +1,183 @@
+                ResourceGroupName = parsedResourceId.ResourceGroupName;
+            }
+
+            PSVirtualHub virtualHubToUpdate = this.GetVirtualHub(this.ResourceGroupName, this.Name);
```
GetVirtualHub will fail if it is not found.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/SetAzureRmVirtualHubCommand.cs:
```diff
@@ -0,0 +1,183 @@
+        DefaultParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+        SupportsShouldProcess = true),
+        OutputType(typeof(PSVirtualHub))]
+    public class SetAzureRmVirtualHubCommand : VirtualHubBaseCmdlet
```
Removed.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/UpdateAzureRmVirtualWanCommand.cs:
```diff
@@ -0,0 +1,157 @@
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+    [Cmdlet("Update",
+        "AzureRmVirtualWan",
```
Fixed

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/UpdateAzureRmVirtualWanCommand.cs:
```diff
@@ -0,0 +1,157 @@
+                this.Name,
+                () =>
+                {
+                    WriteWarning(String.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
Changed to WriteVerbose

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnConnection/GetAzureRmVpnConnectionCommand.cs:
```diff
@@ -0,0 +1,107 @@
+            Mandatory = false,
+            HelpMessage = "The resource name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
```
Fixed

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnConnection/NewAzureRmVpnConnectionCommand.cs:
```diff
@@ -0,0 +1,280 @@
+            Mandatory = true,
+            HelpMessage = "The resource name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
```
Fixed.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnConnection/NewAzureRmVpnConnectionCommand.cs:
```diff
@@ -0,0 +1,280 @@
+                this.Name,
+                () =>
+                {
+                    WriteWarning(String.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
Changed to WriteVerbose

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnConnection/RemoveAzureRmVpnConnectionCommand.cs:
```diff
@@ -0,0 +1,147 @@
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
```
Fixed

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnConnection/RemoveAzureRmVpnConnectionCommand.cs:
```diff
@@ -0,0 +1,147 @@
+                //// At this point, the resource id should not be null. If it is, customer did not specify a valid resource to delete.
+                if (string.IsNullOrWhiteSpace(this.ResourceId))
+                {
+                    throw new PSArgumentException("No vpn connection specified. Nothing will be deleted.");
```
Fixed

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnConnection/UpdateAzureRmVpnConnectionCommand.cs:
```diff
@@ -0,0 +1,179 @@
+                    this.Name,
+                    () =>
+                    {
+                        WriteWarning(String.Format(Properties.Resources.UpdatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
Changed to WriteVerbose

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnConnection/UpdateAzureRmVpnConnectionCommand.cs:
```diff
@@ -0,0 +1,179 @@
+                    () =>
+                    {
+                        WriteWarning(String.Format(Properties.Resources.UpdatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
+                        this.CreateOrUpdateVpnGateway(this.ResourceGroupName, this.ParentResourceName, parentGateway, parentGateway.Tag);
```
The call is made in the context of the parent resource. The connection is modified and added back to the parent resource.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnGateway/GetAzureRmVpnGatewayCommand.cs:
```diff
@@ -0,0 +1,65 @@
+
+    [Cmdlet(VerbsCommon.Get,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VpnGateway",
+        DefaultParameterSetName = "ListByResourceGroupName"),
```
Fixed.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnGateway/NewAzureRmVpnGatewayCommand.cs:
```diff
@@ -0,0 +1,155 @@
+                this.Name,
+                () =>
+                {
+                    WriteWarning(String.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
Changed to WriteVerbose.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnGateway/RemoveAzureRmVpnGatewayCommand.cs:
```diff
@@ -0,0 +1,114 @@
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
```
Fixed

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnGateway/RemoveAzureRmVpnGatewayCommand.cs:
```diff
@@ -0,0 +1,114 @@
+            }
+
+            base.Execute();
+            WriteWarning(Properties.Resources.RemovingVpnGatewayWarning);
```
Changed to WriteVerbose.

maddieclayton replied:
Ping

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnGateway/UpdateAzureRmVpnGatewayCommand.cs:
```diff
@@ -0,0 +1,142 @@
+                    this.Name,
+                    () =>
+                    {
+                        WriteWarning(String.Format(Properties.Resources.UpdatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
Changed to WriteVerbose.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnSite/GetAzureRmVpnSiteCommand.cs:
```diff
@@ -0,0 +1,62 @@
+
+    [Cmdlet(VerbsCommon.Get,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VpnSite",
+        DefaultParameterSetName = "ListByResourceGroupName"),
```
Fixed

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnSite/NewAzureRmVpnSiteCommand.cs:
```diff
@@ -0,0 +1,214 @@
+        public string VirtualWanId { get; set; }
+
+        [Parameter(Mandatory = true,
+            HelpMessage = "The resource name.")]
```
Fixed

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/NewAzureRmVirtualWanCommand.cs:
```diff
@@ -0,0 +1,116 @@
+            var virtualWanModel = NetworkResourceManagerProfile.Mapper.Map<MNM.VirtualWAN>(virtualWan);
+            virtualWanModel.Tags = TagsConversionHelper.CreateTagDictionary(this.Tag, validate: true);
+
+            WriteWarning(string.Format(Properties.Resources.CreatingLongRunningOperationMessage, this.ResourceGroupName, this.Name));
```
Fixed

maddieclayton replied:
Ping, this is still warning.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualHub/GetAzureRmVirtualHubCommand.cs:
```diff
@@ -0,0 +1,62 @@
+
+    [Cmdlet(VerbsCommon.Get,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualHub",
+        DefaultParameterSetName = "ListByResourceGroupName"),
```
Fixed

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/GetAzureRmVirtualWanCommand.cs:
```diff
@@ -0,0 +1,63 @@
+
+    [Cmdlet(VerbsCommon.Get,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualWan",
+        DefaultParameterSetName = "ListByResourceGroupName"),
```
Fixed

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualHub/GetAzureRmVirtualHubCommand.cs:
```diff
@@ -0,0 +1,62 @@
+    {
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = "ListBySubscriptionId",
```
Thanks - applied fix to all gets.

ritwikbasu commented on src/ResourceManager/Network/Commands.Network/help/AzureRM.Network.md:
```diff
@@ -312,7 +317,7 @@ Lists an Azure virtual network gateway's BGP peers
 
 ### [Get-AzureRmVirtualNetworkGatewayConnectionSharedKey](Get-AzureRmVirtualNetworkGatewayConnectionSharedKey.md)
-
+Displays the shared key used for the connection.
```
This is not my change - I just rebased to preview branch.

maddieclayton commented on src/ResourceManager/Network/Commands.Network/Cortex/VpnGateway/GetAzureRmVpnGatewayCommand.cs:
```diff
@@ -0,0 +1,62 @@
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = "ListByResourceGroupName",
+            ValueFromPipelineByPropertyName = true,
```
remove ValueFromPipelineByPropertyName = true

maddieclayton commented on src/ResourceManager/Network/Commands.Network/Cortex/VirtualWan/UpdateAzureRmVirtualWanCommand.cs:
```diff
@@ -0,0 +1,157 @@
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
```
remove this

ritwikbasu replied:
Fixed
